### PR TITLE
Cleanups

### PR DESCRIPTION
**Pull request by tesuji**
* Library crate doesn't need a Cargo.lock file
* lib: Use assert_eq in necessary places
* rustfmt and clippy run
* Add gitignore to ignore Cargo.lock file
* Implement Debug for testing struct

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -236,18 +239,49 @@ mod test {
...
-        assert!(read_array::<Foo>(a) == &[Foo { a: 42 }, Foo { a: 43 }, Foo { a: 44 }, Foo { a: 45 }]);
-
-        let a = &[42, 0, 0, 0, 0x03, 0xff, 0x62, 0xa2, 0x5b, 0x42, 0x00, 0xf0, -2i8 as u8];
-        assert!(read_array::<Bar>(a) == &[Bar { a: 42, b: 0xf000425b_a262ff03, c: -2 }]);
-        let a = &[42, 0, 0, 0, 0x03, 0xff, 0x62, 0xa2, 0x5b, 0x42, 0x00, 0xf0, -2i8 as u8,
-                  43, 0, 0, 0, 0x03, 0xff, 0x62, 0xa2, 0x5b, 0x42, 0x00, 0xf0, -2i8 as u8,
-                  44, 0, 0, 0, 0x03, 0xff, 0x62, 0xa2, 0x5b, 0x42, 0x00, 0xf0, -2i8 as u8];
-        assert!(read_array::<Bar>(a) == &[Bar { a: 42, b: 0xf000425b_a262ff03, c: -2 },
-                                          Bar { a: 43, b: 0xf000425b_a262ff03, c: -2 },
-                                          Bar { a: 44, b: 0xf000425b_a262ff03, c: -2 }]);
```
Comment (nrc) [inline src/lib.rs:250]: I think the formatting of these arrays is useful, so better to use `#[rustfmt::skip]`

**Review by nrc — COMMENTED** [review]
Looks good, thanks! I prefer to leave the Cargo.lock file in though. I think it is useful for libraries to have a working build for developers, even if that is not used by downstream code.

**Comment by tesuji** [comment]
The new lock file would be
```console
% cat Cargo.lock 
[[package]]
name = "zero"
version = "0.1.2"
```

**Comment by tesuji** [comment]
Also to have a working build, I suggest using CI service.

**Comment by tesuji** [comment]
ping @nrc

**Comment by nrc** [comment]
Looks good, thanks for the changes!

**Pull request merged**